src/dehb/utils/config_repository.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import numpy as np
+
+
+@dataclass
+class ConfigItem:
+    config_id: int
+    config: np.array
+    fidelities: dict
+
+@dataclass
+class ResultItem:
+    score: float
+    cost: float
+    info: dict
+
+class ConfigRepository:
+    def __init__(self) -> None:
+        self.configs = []
+
+    def reset(self) -> None:
+        self.configs = []
+
+    def announce_config(self, config: np.array, fidelity: float) -> int:
+        config_id = len(self.configs)
+        result_item = {
+                fidelity: ResultItem(np.inf, -1, {}),
+            }
+        config_item = ConfigItem(config_id, config, result_item)
+        self.configs.append(config_item)
+        return config_id
+
+    def announce_population(self, population: np.array, fidelity=None) -> np.array:
+        """Announce population, retrieving ids for the population.
+
+        Args:
+            population (np.array): Population to announce
+            fidelity (float, optional): Fidelity on which pop is evaluated or None.
+                                        Defaults to None.
+
+        Returns:
+            np.array: population ids
+        """
+        population_ids = []
+        for indiv in population:
+            conf_id = self.announce_config(indiv, float(fidelity or 0))
+            population_ids.append(conf_id)
+        return np.array(population_ids)
+
+    def announce_fidelity(self, config_id: int, fidelity: float):
+        """Announce the evaluation of a new fidelity for a given config.
+
+        This function may only be used if the config already exists in the repository.
+
+        Args:
+            config_id (int): ID of Configuration
+            fidelity (float): Fidelity on which the config will be evaluated
+        """
+        if config_id >= len(self.configs) or config_id < 0:
+            # TODO: Error message
+            return
+
+        config_item = self.configs[config_id]
+        result_item = {
+                fidelity: ResultItem(np.inf, -1, {}),
+            }
+        config_item.fidelities[fidelity] = result_item
```

Comment:
Instead of the if-return above, should be instead return a state such as success/failure/etc.?
Not sure if it messes up how it is being called. 
Just a thought.